### ISSUE TITLE
single recipient conversation ends as group after update to 4.9.0

### ISSUE THREAD

**Issue by mpas97** [WhisperSystems/Signal-Android#6847]
I have:
- [X] searched open and closed issues for duplicates
- [X] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
After updating to beta 4.9.0 on android a single recipient sms conversation turned into an group chat.

### Before
![image](https://user-images.githubusercontent.com/9994218/28873684-fd95e60a-778e-11e7-9ea2-cc5f09c957fe.png)

### After
![image](https://user-images.githubusercontent.com/9994218/28873693-0673f8e8-778f-11e7-857d-981469f47f21.png)

### Device info

**Device:** Samsung Galaxy S6 (tested with an Umi Super too with same result)
**Android version:** 7.0
**Signal version:** 4.9.0

**Comment by arikb** [comment]
Encountering the exact same issue on Signal 4.9.2. [Debug log](https://gist.github.com/anonymous/ea8f4fb2b2f0cafecab1862fd0230447)

This is also on an SMS and it's one of those SMS that come from a non-numeric sender that has a name with a space - of the form "xxx xxxxxx" - so it's similar to the example above in that respect.

SMS from a non-numeric sender where the name of the sender has only one word (no spaces) did not convert to a group conversation.

**Comment by moxie0** [comment]
Interesting, I wasn't expecting these non-numeric identifiers, particularly identifiers with spaces in them. Is it possible (or your expectation) to be able to *send* messages to these identifiers in addition to receiving messages from them?

**Comment by mpas97** [comment]
I have this SMS example only. And this is an ad where I can't answer ( I won't get a reply).  Btw: I just saw in picture before it's a SMS and in the one after it's a MMS.

**Comment by haffenloher** [comment]
Just tried replying to one of those non-numeric senders (without spaces in the name), and to my surprise, the SMS got sent (and I got a reply).

**Comment by haffenloher** [comment]
Just tried replying to one of those non-numeric senders (without spaces). It worked and I got a reply.

**Comment by moxie0** [comment]
The fix will unfortunately not resolve existing threads for people who had already migrated. However, any new messages from "MTV mobile" should show up in their own correctly labeled non-group thread. After you up get the update to 4.9.4, if there's any way to trigger receiving an SMS from these senders in order to verify that, I'd appreciate it. Thanks!

**Comment by Rolf558** [comment]
@moxie0 I got a permanent crash after receiving a SMS of the following format two days ago:

Text:
"Hallo, Ihre mTAN lautet 1234. Ihre Sendung liegt in der PACKSTATION 147 in Kerpen (Sindorfer Str. 65-69, Kaufland). Ihr DHL Team"

Sender:
"DHL"

Note: I have changed the number behind mTAN in the text above.

Unfortunately I cannot provide a log because I immediately uninstalled Signal because it was an important SMS that I asked for re-sending.

I think it was Signal 4.9.2 oder 4.9.3.

Receiving these types of SMS has worked without problems with builds older than 4.9.0.
